### PR TITLE
feat(timeout): expose the connection timeout as an option

### PR DESCRIPTION
**Pull request by joshuachp**
Make the connection timeout configurable to support for slower networks.

### CODE REVIEW

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -0,0 +1,124 @@
+# This file is part of Astarte.
+#
+# Copyright 2022 SECO Mind Srl
```
Comment (harlem88) [inline .github/workflows/check.yaml:3]:
```suggestion
# Copyright 2024 SECO Mind Srl
```

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,104 @@
+# This file is part of Astarte.
+# Copyright 2023 SECO Mind Srl
```
Comment (harlem88) [inline .github/workflows/test.yaml:2]:
```suggestion
# This file is part of Astarte.
#
# Copyright 2024 SECO Mind Srl
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: .github/workflows/check.yaml (hunk 1 above)
Comment (joshuachp) [reply .github/workflows/check.yaml]: Done

File: .github/workflows/test.yaml (hunk 2 above)
Comment (joshuachp) [reply .github/workflows/test.yaml]: Done

**Review by harlem88 — APPROVED** [review]

**Review by rgallor — APPROVED** [review]

**Pull request merged**